frontend/front/src/pages/Surveyor/account/Account.js

```diff
@@ -3,42 +3,69 @@ import React from "react";
 import AccountDetail from "./AccountDetail";
 import { Link } from "react-router-dom";
 import { useSelector } from "react-redux";
+import { selectCurrentUser } from "../../../features/login/loginSlice";
+import { useGetSurveyorQuery } from "../../../api/apiSlice";
+import Loader from "../../../components/Loader";
+import CustomSnackbar from "../../../components/CustomSnackbar";
 
 const Account = () => {
-  const { firstName, lastName, email, address, phoneNumber } = useSelector(
-    (state) => state.account
-  );
+  const { id: userId } = useSelector(selectCurrentUser);
```

Comment:
This won't work in practice, because the current user contains the _user_ id, but what you want is the _surveyor_ id, and they aren't guaranteed to be the same. #399 adds the surveyor email to the user state, and you should be able to use that to do like `/surveyors?email=${email}` to get the user data.

Reply:
Actually, should I just have it put the whole surveyor in the JWT?

Reply:
What would be the advantage of storing the whole surveyor in the JWT? The solution in your previous comment makes sense to me.

Reply:
I was thinking then you'd have the surveyor id itself

Reply:
note to self: select surveyor_id (or similar) instead of just id

Reply:
@mattdelsordo When I curl `localhost:3000/surveyors?email=admin@heat.pump` or actually any params in that format (`/surveyors?id=1` etc.) I just get back the whole `surveyors.json` every time. We were running into a similar thing with getting unassigned homes with `/homes?assignment=null`. Maybe these param requests need to be explicitly set up?

Reply:
I dug a bit - it's in the controller.
```
def search params
  params.permit(:param1, :param2, etc.)
```

Reply:
This is resolved in commit f34e333

Reply:
Sorry to re-open this but with #399 you should be able to get the `surveyor_id` out of the user object directly! So you don't actually need to look up with the email after all.